<?php

declare(strict_types=1);

/*
 * This file has been auto generated by Jane,
 *
 * Do no edit it directly.
 */

namespace CedricZiel\Baserow\Generated\Exception;

class AsyncAuditLogExport2NotFoundException extends NotFoundException
{
    /**
     * @var \CedricZiel\Baserow\Generated\Model\ApiAuditLogExportPostResponse404
     */
    private $apiAuditLogExportPostResponse404;
    /**
     * @var \Psr\Http\Message\ResponseInterface
     */
    private $response;

    public function __construct(\CedricZiel\Baserow\Generated\Model\ApiAuditLogExportPostResponse404 $apiAuditLogExportPostResponse404, \Psr\Http\Message\ResponseInterface $response)
    {
        parent::__construct('Not Found');
        $this->apiAuditLogExportPostResponse404 = $apiAuditLogExportPostResponse404;
        $this->response = $response;
    }

    public function getApiAuditLogExportPostResponse404(): \CedricZiel\Baserow\Generated\Model\ApiAuditLogExportPostResponse404
    {
        return $this->apiAuditLogExportPostResponse404;
    }

    public function getResponse(): \Psr\Http\Message\ResponseInterface
    {
        return $this->response;
    }
}
